### DISCUSSION TITLE
programming error:Failed to create fpgenprog project

### DISCUSSION THREAD

**Discussion by xing-git** [polarfire-soc/polarfire-soc-documentation#137] [HSS and U-Boot] [answered]
The softconsole version is **2021.3**. build this project on **windows**. added fpgenprog environment.

When I build project,build info is:
ERROR - **Failed to create fpgenprog project** at location "D:\code\hart-software-services-master\Default/bootmode1/fpgenprogProject".
d:/microchip/softconsole-v2021.3-7.0.0.599/eclipse/../build_tools/bin/sh: D:/Microchip/SoftConsole-v2021.3-7.0.0.599//eclipse/jre/bin/java.exe: No error
cp: can't stat 'Default/bootmode1/hss-envm-wrapper-bm1-p0.hex': No such file or directory

I want to know if I need someother configs or do something, thanks

**Comment by hughbreslin** [comment, marked as answer]
Hey @xing-git 

I have a few questions on this:
1. Can you see the `hss-l2-scratch.elf` in the `Default` folder? I.e has the HSS build succeeded? 
2. What version of the HSS is this and what version of Libero/Program and Debug tools are you using?
3. How did you add the `fpgenprog` environment variable?
4. Have you tried building from your `C` drive with the workspace located there? I haven't tried building from an external drive, in your case I can see its `D` 

You shouldn't need any additional configuration to generate the boot mode 1 configuration - the main thing is the `fpgenprog` tool

**Reply by xing-git** [reply]
1.HSS build secceed except hss-envm-wrapper-bm1-p0.hex,this is Default folder:
![image](https://user-images.githubusercontent.com/20128646/179647036-540f8315-690c-4cdc-a1d9-05a5d7e5ebb5.png)
  but in bootmode1 folder ,thers is only debugLog.txt.
2.HSS is the latest version,download at 2022-7-2; LIbero version is 2021.1.0.17
3.In softconsole,add environment variable"FPGENPROG" and the value is fpgenprog.exe path.
![image](https://user-images.githubusercontent.com/20128646/179648109-f9ee2faa-e6a5-4738-a449-53e9a8fffc57.png)
4.Have tried copy project to C and build in C ,but same as above

Now I can run the HSS by FlashPro Debug,but I want to write HSS to eNVM,then I need a .hex file.

**Reply by AntonKrug** [reply]
The `FPGENPROG` value doesn't look right, see this section:
https://mi-v-ecosystem.github.io/SoftConsole-Documentation/SoftConsole-v2021.3/using_softconsole/mpfs.html#fpgenprog

**Reply by xing-git** [reply]
In my notebook,I dont install Libero,so I copy fpgenprog.exe into my project folder an add the path to Environment.
But in another PC which installed libero,the environment is like this,but not work as well.
![捕获](https://user-images.githubusercontent.com/20128646/179919594-dd48e94d-810f-44ac-b163-094e58a96577.JPG)

**Reply by hughbreslin** [reply]
From the link @AntonKrug shared, you need to enter the full path to `FPGENPROG` application, e.g:

> D:\Microsemi\Libero_SoC_v2021.1\Designer\bin64\fpgenprog.exe

**Reply by xing-git** [reply]
thanks for @AntonKrug  and @hughbreslin ,it works when I enter full path.
